src/main/java/com/dislinkt/post/service/PostService.java

```diff
@@ -0,0 +1,84 @@
+package com.dislinkt.post.service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import com.dislinkt.post.dto.PostDTO;
+import com.dislinkt.post.mapper.PostMapper;
+import com.dislinkt.post.model.Person;
+import com.dislinkt.post.model.Post;
+import com.dislinkt.post.repository.PostRepository;
+
+@Service
+public class PostService {
+
+    @Autowired
+    private PostRepository repository;
+
+    @Autowired
+    private PersonService personService;
+
+    private PostMapper mapper = new PostMapper();
+
+    public List<PostDTO> findAll(){
+        return mapper.toDtoList(repository.findAll());
+    }
+
+    public List<PostDTO> findByPersonId(UUID personId) throws Exception{
+        Person person = personService.findOne(personId);
+        if (person == null)
+            throw new Exception("User with given id doesn't exist!");
+
+        List<Post> ret = new ArrayList<>();
+        for (Post p: person.getPosts())
+            ret.add(p);
```

Comment:
I don't think you have to call personService here at all.
You can just query PostRepository by personId and you don't have to care about a person's existence, because if it doesn't exist you'll just have an empty list. What do you think?

Reply:
I just simplified it. Thank you for the suggestion.